File: app/src/main/java/com/task/usecase/UseCase.java
package com.task.usecase;

import com.task.data.remote.dto.NewsItem;
import com.task.ui.base.listeners.BaseCallback;

import java.util.List;

/**
 * Created by ahmedeltaher on 3/22/17.
 */

public interface UseCase {
    void getNews(final BaseCallback callback);

    NewsItem searchByTitle(List<NewsItem> news, String keyWord);
}

File: app/src/main/java/com/task/usecase/UseCase.kt
package com.task.usecase

import androidx.lifecycle.MutableLiveData
import com.task.data.Resource
import com.task.data.remote.dto.NewsItem
import com.task.data.remote.dto.NewsModel

/**
 * Created by AhmedEltaher
 */

interface UseCase {
    fun getNews()
    fun searchByTitle(keyWord: String): NewsItem?
    val newsLiveData: MutableLiveData<Resource<NewsModel>>
}
